legion/legion/k8s/services.py

```diff
@@ -371,13 +371,33 @@ def delete(self, grace_period_seconds=0):
         core_v1api.delete_namespaced_service(name=self.k8s_service.metadata.name, body=body,
                                              namespace=self.namespace)
 
+        service_deleted = ensure_function_succeed(
+            lambda: self.k8s_service.metadata.name not in [
+                item.metadata.name
+                for item in core_v1api.list_namespaced_service(self.namespace).items],
+            10, 1, boolean_check=True
+        )
+
+        if not service_deleted:
+            raise Exception('Cannot delete service')
```

Comment:
@kirillmakhonin use custom exception type to make custom handling logic possible in future